### PR TITLE
test: Use chalk for output styling

### PR DESCRIPTION
**Pull request by ExE-Boss**
Opened because of @lahmatiy’s review in https://github.com/mdn/data/pull/362#discussion_r274075034.

## Depends on:
- [x] ~~#4036~~ (merged)
- [x] ~~#4047~~ (merged)
- [x] ~~#4078~~ (merged)

---

review?(@Elchi3)

### CODE REVIEW

**Comment by vinyldarkscratch** [comment]
First off, I'm in favor of using `chalk` for color coding output, since it's a little more self-documenting.  However, there are several changes in here that don't relate to switching to `chalk`, such as formatting, syntax updates, and possibly the upgrades of some of the packages.  Can we separate those changes into other PRs?

**Comment by Elchi3** [comment]
I agree. `chalk` might be nice, but this PR does too many things at once. If they are prerequisites for `chalk`, you might want to file PRs for them first.

**Comment by ExE-Boss** [comment]
The issue with `chalk.reset` is&nbsp;that it&nbsp;wraps the&nbsp;text argument (which mustn't be&nbsp;the&nbsp;empty string) in&nbsp;`\x1b[0m` on&nbsp;both sides, which is&nbsp;unwanted.

**Comment by vinyldarkscratch** [comment]
From my testing, that `\x1b[0m` doesn't actually do anything.  We can safely remove it without altering the output.

Screen Shot 2019-05-05 at 14 52 37

Alternatively, to bring back the old styling behavior, we can throw `chalk.reset()` around the two lines showing the actual/expected outputs.

```js
/**
 * @param {string} actual
 * @param {string} expected
 * @return {string}
 */
function jsonDiff(actual, expected) {
  var actualLines = actual.split(/\n/);
  var expectedLines = expected.split(/\n/);

  for (var i = 0; i < actualLines.length; i++) {
    if (actualLines[i] !== expectedLines[i]) {
      return [
        '#' + (i + 1),
        chalk.reset('    Actual:   ' + escapeInvisibles(actualLines[i])),
        chalk.reset('    Expected: ' + escapeInvisibles(expectedLines[i])),
      ].join('\n');
    }
  }
}
```

File: test/test-style.js [hunk 1]
```diff
@@ -113,11 +114,9 @@ function jsonDiff(actual, expected) {
 
   for (var i = 0; i < actualLines.length; i++) {
     if (actualLines[i] !== expectedLines[i]) {
-      return [
-        '#' + (i + 1) + '\x1b[0m',
-        '    Actual:   ' + escapeInvisibles(actualLines[i]),
-        '    Expected: ' + escapeInvisibles(expectedLines[i]),
-      ].join('\n');
+      return chalk`#${i + 1}{reset
```
Comment (vinyldarkscratch) [inline test/test-style.js]: I'm a fan of using the string templating when we can, but this is inconsistent with the rest of the files.  We should probably choose one and stick with it (and I'd say that the templating method would be better), but I think that this is a decision we can make later.

Comment (ExE-Boss) [reply test/test-style.js]: This is because currently, only the top&#x2011;level `chalk` function can be&nbsp;a&nbsp;template function: https://github.com/chalk/chalk/issues/341

**Comment by Elchi3** [comment]
This commit https://github.com/mdn/browser-compat-data/pull/4015/commits/042484d5e8189c1e26c65df6c11d450693e31ffd has more fixes than the use of chalk. So more to review. I think it is a different bug that is now addressed in this PR for some reason. You can tell me what it is or I will now spend time to figure it out. In any case, it would be better to do such things in separate PRs (I said this before in other situations).

File: scripts/release-notes.js [hunk 2]
```diff
@@ -1,12 +1,13 @@
...
 const http = require('https');
 const readline = require('readline');
+const chalk = require('chalk');
 
 const bcd = require('..');
 
 const { argv } = require('yargs')
   .command('$0 <version-tag>', 'Initiate a release of this package on GitHub', (yargs) => {
-    yargs.positional('version-tag', {
+    return yargs.positional('version-tag', {
```
Comment (vinyldarkscratch) [inline scripts/release-notes.js]: This doesn't seem like a `chalk` change either (don't know why I didn't notice it before, whoops).  What's the purpose of this, and can we move this into a separate PR?

File: test/test-style.js [hunk 3]
```diff
@@ -128,39 +127,42 @@ function jsonDiff(actual, expected) {
 function testStyle(filename) {
   let hasErrors = false;
   let actual = fs.readFileSync(filename, 'utf-8').trim();
-  let expected = JSON.stringify(JSON.parse(actual), null, 2);
+  /** @type {import('../types').CompatData} */
+  let dataObject = JSON.parse(actual);
+  let expected = JSON.stringify(dataObject, null, 2);
+  let expectedSorting = JSON.stringify(dataObject, orderSupportBlock, 2);
```
Comment (vinyldarkscratch) [inline test/test-style.js]: Just got all the non-chalk changes in here, and then I’d say this PR is ready to go!

Comment (ExE-Boss) [reply test/test-style.js]: Moved to #4078

File: scripts/release-notes.js (hunk 2 above)
Comment (ExE-Boss) [reply scripts/release-notes.js]: This is actually a bug in DefinitelyTyped: https://github.com/DefinitelyTyped/DefinitelyTyped/pull/35234.

**Comment by ExE-Boss** [comment]
review?(@Elchi3): I’ve now addressed all the dependencies of this and this is ready for review.

**Comment by Elchi3** [comment]
Yes, thank you @ExE-Boss for taking the time to separate out additional changes. I wasn't asking to do that to give you additional work or to annoy you, but PRs that go beyond their initial scope (or have multiple scopes in the first place) are really hard to review and don't communicate well. 

I believe we now have a better result actually. If you look at the depended PRs, you see that by looking at issues independently we were able review them in detail and came up with even more improvements along our way for the specific issues you've found. I think this is awesome and is the super-power of collaboration and talking to each other about changes.

Also, in the future, we can refer to these separate PRs as git blame will point us to them. If said PRs then also come with clear intent and communication why some things where changed, future and present contributors and reviewers to this project have a chance to participate as understanding things and finding reasoning is enormously easier.

To me, communication around changes is often more important than the change itself.

Thanks again for your work.
